src/intervals/construction.jl

```diff
@@ -259,31 +253,275 @@ function atomic(::Type{T}, a::AbstractFloat) where {T<:AbstractFloat}
+
+#
+
+function interval(::Type{T}, a, b, d::Decoration; format::Symbol = :standard) where {T<:NumTypes}
+    format === :standard && return _interval_standard(T, a, b, d)
+    format === :midpoint && return _interval_midpoint(T, a, b, d)
+    return throw(ArgumentError("`format` must be `:standard` or `:midpoint`."))
+end
+interval(a, b, d::Decoration; format::Symbol = :standard) = interval(promote_numtype(numtype(a), numtype(b)), a, b, d; format = format)
+
+function interval(::Type{T}, a, b; format::Symbol = :standard) where {T<:NumTypes}
+    format === :standard && return _interval_standard(T, a, b)
+    format === :midpoint && return _interval_midpoint(T, a, b)
+    return throw(ArgumentError("`format` must be `:standard` or `:midpoint`."))
+end
+interval(a, b; format::Symbol = :standard) = interval(promote_numtype(numtype(a), numtype(b)), a, b; format = format)
+
+function interval(::Type{T}, a, d::Decoration; format::Symbol = :standard) where {T<:NumTypes}
+    ((format === :standard) | (format === :midpoint)) && return _interval_standard(T, a, a, d)
+    return throw(ArgumentError("`format` must be `:standard` or `:midpoint`."))
+end
+function interval(::Type{T}, x::Union{BareInterval,Interval}, d::Decoration; format::Symbol = :standard) where {T<:NumTypes}
+    ((format === :standard) | (format === :midpoint)) && return _unsafe_interval(bareinterval(T, x), min(decoration(x), d), isguaranteed(x)) # assumes valid interval
+    return throw(ArgumentError("`format` must be `:standard` or `:midpoint`."))
+end
+interval(a, d::Decoration; format::Symbol = :standard) = interval(promote_numtype(numtype(a), numtype(a)), a, d; format = format)
+
+function interval(::Type{T}, a; format::Symbol = :standard) where {T<:NumTypes}
+    ((format === :standard) | (format === :midpoint)) && return _interval_standard(T, a, a)
+    return throw(ArgumentError("`format` must be `:standard` or `:midpoint`."))
+end
+function interval(::Type{T}, x::Union{BareInterval,Interval}; format::Symbol = :standard) where {T<:NumTypes}
+    ((format === :standard) | (format === :midpoint)) && return _unsafe_interval(bareinterval(T, x), decoration(x), isguaranteed(x)) # assumes valid interval
+    return throw(ArgumentError("`format` must be `:standard` or `:midpoint`."))
+end
+interval(a; format::Symbol = :standard) = interval(promote_numtype(numtype(a), numtype(a)), a; format = format)
+
+# some useful extra constructor
+interval(::Type{T}, a::Tuple, d::Decoration; format::Symbol = :standard) where {T<:NumTypes} = interval(T, a..., d; format = format)
+interval(a::Tuple, d::Decoration; format::Symbol = :standard) = interval(a..., d; format = format)
+interval(::Type{T}, a::Tuple; format::Symbol = :standard) where {T<:NumTypes} = interval(T, a...; format = format)
+interval(a::Tuple; format::Symbol = :standard) = interval(a...; format = format)
+
+# standard format
+
+function _interval_standard(::Type{T}, a, b, d::Decoration) where {T<:NumTypes}
```

Comment:
Docstrings for `_interval_standard` seem missing.

Reply:
Done.